src/cclib/parser/ccio.py

```diff
@@ -136,20 +137,29 @@ def ccopen(source, *args, **kargs):
         except IOError as error:
             if not kargs.get('quiet', False):
                 (errno, strerror) = error.args
-                print("I/O error %s (%s): %s" % (errno, source, strerror))
             return None
     elif hasattr(source, "read"):
         inputfile = source
         isstream = True
 
+    if isstream:
+        try:
+            inputfile.seek(0, 0)
+        except (AttributeError, IOError):
+            contents = inputfile.read()
+            inputfile = io.StringIO(contents)
+            inputfile.seek(0, 0)
+
     # Proceed to return an instance of the logfile parser only if the filetype
     # could be guessed. Need to make sure the input file is closed before creating
     # an instance, because parsers will handle opening/closing on their own.
     filetype = guess_filetype(inputfile)
+    inputfile.seek(0, 0)
```

Comment:
If you call ```inputfile.seek(0, 0)``` here, why do you need a special case for ```isstream``` other than for the try-except block? Seems like you could get rid of the ```if istream``` block and remove the extra seek here.